onnx/defs/nn/defs.cc

```diff
@@ -651,6 +651,9 @@ output_shape can also be explicitly specified in which case pads values are auto
   If (auto_pads != SAME_UPPER): pads[start_i] = total_padding[i]/2; pads[end_i] = total_padding[i] - (total_padding[i]/2)
   Else: pads[start_i] = total_padding[i] - (total_padding[i]/2); pads[end_i] = (total_padding[i]/2).
 
+Notice although we define parameter pads as the input padding,
+for ease of calculation, we used following transformation for given input padding pads' before calculation:
+  pads = pads' - k - 1
```

Comment:
Sorry I don't quite understand this line, could you elaborate?

Reply:
@houseroad and I discussed about our definition of "output_pads" and "pads", and found that both of them are output padding, although we claim this "pads" is input padding. In order to translate this "pads" to input padding, we need this formula. 
See http://deeplearning.net/software/theano/tutorial/conv_arithmetic.html#transposed-convolution-arithmetic Relationship 8